neqo-crypto/tests/agent.rs

```diff
@@ -535,3 +536,100 @@ fn ech_retry() {
     assert!(client.preinfo().unwrap().ech_accepted().unwrap());
     assert!(server.preinfo().unwrap().ech_accepted().unwrap());
 }
+
+#[test]
+fn connection_succeeds_when_server_and_client_support_cert_compr_copy() {
+    struct CopyCompression {}
+
+    // Implementation supports both encoder and decoder
+    impl SafeCertificateCompression for CopyCompression {
+        const ID: u16 = 0x4;
+        const NAME: &str = concat!("copy", "\0");
+        const ENABLE_ENCODING: bool = true;
+
+        fn encode(data: &[u8]) -> Vec<u8> {
+            data.to_vec()
+        }
+        fn decode(data: &[u8]) -> Vec<u8> {
+            data.to_vec()
+        }
+    }
+
+    fixture_init();
+    let mut client = Client::new("server.example", true).expect("should create client");
+    let mut server = Server::new(&["key"]).expect("should create server");
+
+    client
+        .set_certificate_compression::<CopyCompression>()
+        .unwrap();
+    server
+        .set_certificate_compression::<CopyCompression>()
+        .unwrap();
+
+    connect(&mut client, &mut server);
+
+    assert!(client.state().is_connected());
+    assert!(server.state().is_connected());
+}
+
+struct CopyCompressionNoEncoder {}
+
+impl SafeCertificateCompression for CopyCompressionNoEncoder {
+    const ID: u16 = 0x4;
+    const NAME: &str = concat!("copy", "\0");
```

Comment:
The trailing null here should not be necessary!  See above comments about CString usage.  It involves allocation, but that's fine in this case (and the compiler is likely able to optimize that out for this).